pkg/controller/service/servicecontroller.go

```diff
@@ -581,9 +581,15 @@ func stringSlicesEqual(x, y []string) bool {
 }
 
 func hostsFromNodeList(list *api.NodeList) []string {
-	result := make([]string, len(list.Items))
+	result := []string{}
 	for ix := range list.Items {
-		result[ix] = list.Items[ix].Name
+		if list.Items[ix].Spec.Unschedulable {
+			continue
+		}
+		if list.Items[ix].Status.Phase != api.NodeRunning {
```

Comment:
heh, I ran into the same problem with L7. you want not ready also no, just use the condition lister https://github.com/bprashanth/contrib/commit/4dc30b3de54dbdf9be9a978589e854d577231ffa#diff-315afc43468bd4edd63930333434d306R358 ?